Asterix/optics/coronagraph.py

```diff
@@ -126,6 +124,60 @@ def __init__(self, modelconfig, coroconfig, Model_local_dir=None):
 
         self.string_os += '_LS' + self.lyot_pup.string_os
 
+        if self.prop_apod2lyot in ['mft', 'mft-babinet']:
+
+            # we measure the AA and BB matrix and norm0 for all MFTs used in coronagraphy
+            # TODO : in practice, MFT direct in case self.AA_directdirect is the same one than self.AA_direct_final
+            if self.prop_apod2lyot == 'mft':
+                dim_science_here = self.dimScience
+                fpm_sampling_here = self.Science_sampling
+            if self.prop_apod2lyot == 'mft-babinet':
+                dim_science_here = self.dim_fpm
+                fpm_sampling_here = self.Lyot_fpm_sampling
+
+            self.AA_direct = list()
+            self.BB_direct = list()
+            self.norm0_direct = list()
+
+            self.AA_inverse = list()
+            self.BB_inverse = list()
+            self.norm0_inverse = list()
```

Comment:
If you are concerned about speed, you might consider initializing lists with brackets:
https://stackoverflow.com/questions/5790860/and-vs-list-and-dict-which-is-better

I repeated the test of the first answer in the link in my asterix conda env on my Mac and got essentially the same results (just tried it because the Stackoverflow question in the link was posted 11 years ago):
```python
In [1]: from timeit import timeit

In [2]: timeit("[]")
Out[2]: 0.03909633299999982

In [3]: timeit("list()")
Out[3]: 0.09334029099999874

In [4]: timeit("{}")
Out[4]: 0.042814207999999354

In [5]: timeit("dict()")
Out[5]: 0.10233649999999983
```

Reply:
ok, but not really important here, since this is done only once at coronagraph initialization

Reply:
bu I coorected anyway